### PR TITLE
Refactor enrichment using Pydantic model validator

### PR DESCRIPTION
**Pull request by disrupted**

### CODE REVIEW

File: kpops/components/base_components/base_defaults_component.py [hunk 1]
```diff
@@ -75,12 +75,20 @@ class BaseDefaultsComponent(DescConfigModel, ABC):
...
 
-    def __init__(self, **kwargs) -> None:
-        if kwargs.get("enrich", True):
-            kwargs = self.extend_with_defaults(**kwargs)
-        super().__init__(**kwargs)
-        if kwargs.get("validate", True):
-            self._validate_custom(**kwargs)
+    @pydantic.model_validator(mode="before")
+    @classmethod
+    def do_enrich(cls, values: Any) -> dict[str, Any]:
```
Comment (disrupted) [inline kpops/components/base_components/base_defaults_component.py:80]: not sure about the function name, `enrich` is already used as attribute

File: kpops/components/base_components/base_defaults_component.py [hunk 2]
```diff
@@ -75,12 +75,20 @@ class BaseDefaultsComponent(DescConfigModel, ABC):
...
+        if not isinstance(values, dict):
+            return {}
+        if values.get("enrich", True):
+            values = cls.extend_with_defaults(**values)
+        return values
+
+    @pydantic.model_validator(mode="after")
+    def do_validate(self) -> Self:
```
Comment (disrupted) [inline kpops/components/base_components/base_defaults_component.py:88]: same with this one

File: kpops/components/base_components/base_defaults_component.py (hunk 1 above)
Comment (disrupted) [reply kpops/components/base_components/base_defaults_component.py]: renamed

File: kpops/utils/gen_schema.py [hunk 3]
```diff
@@ -137,7 +137,10 @@ def gen_pipeline_schema(
             default=component.type,
         )
         core_schema: DefinitionsSchema = component.__pydantic_core_schema__  # pyright:ignore[reportGeneralTypeIssues]
-        model_schema: ModelFieldsSchema = core_schema["schema"]["schema"]  # pyright:ignore[reportGeneralTypeIssues,reportTypedDictNotRequiredAccess]
+
+        model_schema: ModelFieldsSchema = core_schema["schema"]["schema"]["schema"][  # pyright:ignore[reportGeneralTypeIssues,reportTypedDictNotRequiredAccess]
```
Comment (disrupted) [inline kpops/utils/gen_schema.py:141]: not sure why this got nested even more, but that's Pydantic internals 🤷🏻

**Review by sujuka99 — APPROVED** [review]
LGTM

File: poetry.lock
Comment (raminqaf) [inline poetry.lock:1]: Why did the poetry lock changed?

Comment (disrupted) [reply poetry.lock:1]: because I updated Pydantic https://github.com/bakdata/kpops/pull/444/commits/b368f2389405903e238318a5ec7b99d1182647a7

**Pull request merged**